blocks/bricks/interfaces.py

```diff
@@ -127,44 +127,55 @@ class Initializable(RNGMixin, Brick):
-        ``False`` if the brick does not support biases, and only has
-        :attr:`weights_init`.  For an example of this, see
-        :class:`.Bidirectional`. If this is ``False``, the brick does not
-        support the arguments ``biases_init`` or ``use_bias``.
-
     """
-    has_biases = True
 
     @lazy()
-    def __init__(self, weights_init=None, biases_init=None, use_bias=True,
+    def __init__(self, initialization_schemes=None, use_bias=True,
                  seed=None, **kwargs):
-        super(Initializable, self).__init__(**kwargs)
-        self.weights_init = weights_init
-        if self.has_biases:
-            self.biases_init = biases_init
-        elif biases_init is not None or not use_bias:
-            raise ValueError("This brick does not support biases config")
         self.use_bias = use_bias
         self.seed = seed
+        self.initialization_schemes = initialization_schemes
+        self.parameter_roles = set([])
+        if self.initialization_schemes is None:
+            self.initialization_schemes = {}
+
+        kwargs_ = {}
+        for key in kwargs:
+            if key[-5:] == "_init":
+                if key in self.initialization_schemes:
+                    raise ValueError("All initializations are accepted either"
+                                     "through initialization_schemes or "
+                                     "correspodong attribute but not both")
+                else:
+                    self.initialization_schemes[key] = kwargs[key]
+            else:
+                kwargs_[key] = kwargs[key]
+
+        super(Initializable, self).__init__(**kwargs_)
+        self._collect_roles()
 
     def _push_initialization_config(self):
         for child in self.children:
+            if (isinstance(child, Initializable) and
+                    hasattr(child, 'initialization_schemes')):
+                for role in child.initialization_schemes:
+                    if role not in self.parameter_roles:
+                        raise ValueError("The parameter role: " +
+                                         "{} is not defined in".format(role) +
+                                         "in the class parameter_roles")
+
+        for child in self.children:
             if isinstance(child, Initializable):
                 child.rng = self.rng
-                if self.weights_init:
-                    child.weights_init = self.weights_init
-        if hasattr(self, 'biases_init') and self.biases_init:
-            for child in self.children:
-                if (isinstance(child, Initializable) and
-                        hasattr(child, 'biases_init')):
-                    child.biases_init = self.biases_init
+                child.initialization_schemes = self.initialization_schemes
+
+    def _collect_roles(self):
+        for child in self.children:
+            if isinstance(child, Initializable):
+                self.parameter_roles.update(child.parameter_roles)
```

Comment:
I guess it's not necessary if called from the constructor (though I think that might be suboptimal)